### PR TITLE
Refactor getMessage command and Add ability to pin messages

### PR DESCRIPTION
**Pull request by binaryoverload**

### CODE REVIEW

File: src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java [hunk 1]
```diff
@@ -12,11 +13,17 @@
 
     @Override
     public void onCommand(User sender, TextChannel channel, Message message, String[] args, Member member) {
-        if (args.length == 1) {
+        if (args.length == 1 && !args[0].replaceAll("[^0-9]", "").isEmpty()) {
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java]: `args[0].matches("[0-9]{18,20}");`

File: src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java [hunk 2]
```diff
@@ -12,11 +13,17 @@
...
             String messageId = args[0].replaceAll("[^0-9]", "");
             Message msg = channel.getMessageById(messageId).complete();
             msg.pin().complete();
             channel.getHistory().retrievePast(1).complete().get(0).delete().queue();
+        } else if (args.length != 0){
+            String pinMessage = MessageUtils.getMessage(args, 0);
+            Message msg = channel.sendMessage(new EmbedBuilder().setTitle(sender.getName(), null)
+                    .setThumbnail(MessageUtils.getAvatar(sender)).setDescription(pinMessage).build()).complete();
+            msg.pin().complete();
```
Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java]: As usual, you made single use variables

Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java]: Ok inline

File: src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java (hunk 1 above)
Comment (binaryoverload) [inline src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java]: I take it that is the length of a message id?

Comment (ArsenArsen) [inline src/main/java/stream/flarebot/flarebot/commands/administrator/PinCommand.java]: Its a length of a possible snowflake, so yes.

**Comment by binaryoverload** [comment]
Yay!

**Comment by binaryoverload** [comment]
Thnx guys

**Pull request merged**